### PR TITLE
chore: dependencies upgrade

### PR DESCRIPTION
**Pull request by trOnk12**
## Description

In order to make the app up-to-date with the newest goodies, we had to upgrade the dependencies. 

This has following consequences:

1. Custom rule provider had to be adjusted to the newest breaking changes.
2. Updating to newest Gradle and AGP, we had to update java to minimum version of 17.
3. Newest AGP required some change to namespace definition.

### Why?
*Describe why the change is introduced*

## Links to related issues

- Fixes [LD-XXX](https://loudius.atlassian.net/jira/software/projects/LD/boards/3)
- Not covered by the PR: [LD-XXX](https://loudius.atlassian.net/jira/software/projects/LD/boards/3)
- Possible improvements: [LD-XXX](https://loudius.atlassian.net/jira/software/projects/LD/boards/3)

## Demo

| Before the PR | After the PR |
|---------------|--------------|
|               |              |

## Documentation
- Link to documentation: **Add link here**
- Link to figma: **Add link here**
- **Add here other useful documentation links that are useful for PR reviewer**

## Checklist

- [ ] New functionality is covered by Analytics events
- [ ] Analytics events are documented
- [ ] Analytics dashboard is updated
- [ ] Functionality covered by unit tests
- [ ] I have manually tested if the code works

### CODE REVIEW

File: custom-ktlint-rules/build.gradle [hunk 1]
```diff
@@ -4,16 +4,21 @@ plugins {
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(8)
```
Comment (trOnk12) [inline custom-ktlint-rules/build.gradle]: // somehow megalinter has problem building when the java version is above 11,
        // also I was not able to specify 11 by toolchain, so I used 8

File: custom-ktlint-rules/build.gradle [hunk 2]
```diff
@@ -4,16 +4,21 @@ plugins {
...
-    compileOnly 'com.pinterest.ktlint:ktlint-core:0.48.2'
-    compileOnly 'com.pinterest.ktlint:ktlint-ruleset-standard:0.48.2'
+    compileOnly 'com.pinterest.ktlint:ktlint-rule-engine-core:1.0.1'
+    compileOnly 'com.pinterest.ktlint:ktlint-ruleset-standard:1.0.1'
     testImplementation 'junit:junit:4.13.2'
-    testImplementation 'io.strikt:strikt-core:0.34.0'
-    testImplementation 'org.assertj:assertj-core:3.10.0'
-    testImplementation 'com.pinterest.ktlint:ktlint-core:0.48.2'
-    testImplementation 'com.pinterest.ktlint:ktlint-test:0.48.2'
+    implementation 'org.slf4j:slf4j-api:2.0.7'
```
Comment (trOnk12) [inline custom-ktlint-rules/build.gradle:16]: was necessary otherwise linter was failing with NoClassFoundError

File: LICENSE [hunk 3]
```diff
@@ -1,4 +1,3 @@
-                                 Apache License
```
Comment (nowakweronika) [inline LICENSE]: Question: Why did you remove this header?

File: app-shared-tests/build.gradle
Comment (nowakweronika) [inline app-shared-tests/build.gradle:1]: Praise: It's nice to have newer versions of libraries :)

**Review by nowakweronika — COMMENTED** [review]
I can't sync project with gradle files. Is it okay on your device? I got this error message:
```
org.gradle.api.internal.provider.AbstractProperty$PropertyQueryException: Failed to calculate the value of task ':custom-ktlint-rules:compileJava' property 'javaCompiler'.
.
.
.
Caused by: org.gradle.jvm.toolchain.internal.NoToolchainAvailableException: No matching toolchains found for requested specification: {languageVersion=8, vendor=any, implementation=vendor-specific}.
.
.
.
Caused by: org.gradle.jvm.toolchain.internal.ToolchainDownloadFailedException: No locally installed toolchains match (see https://docs.gradle.org/8.1/userguide/toolchains.html#sec:auto_detection) and toolchain download repositories have not been configured (see https://docs.gradle.org/8.1/userguide/toolchains.html#sub:download_repositories).
.
.
.
```

**Comment by trOnk12** [comment]
> I can't sync project with gradle files. Is it okay on your device? I got this error message:
> 
> ```
> org.gradle.api.internal.provider.AbstractProperty$PropertyQueryException: Failed to calculate the value of task ':custom-ktlint-rules:compileJava' property 'javaCompiler'.
> .
> .
> .
> Caused by: org.gradle.jvm.toolchain.internal.NoToolchainAvailableException: No matching toolchains found for requested specification: {languageVersion=8, vendor=any, implementation=vendor-specific}.
> .
> .
> .
> Caused by: org.gradle.jvm.toolchain.internal.ToolchainDownloadFailedException: No locally installed toolchains match (see https://docs.gradle.org/8.1/userguide/toolchains.html#sec:auto_detection) and toolchain download repositories have not been configured (see https://docs.gradle.org/8.1/userguide/toolchains.html#sub:download_repositories).
> .
> .
> .
> ```

damn that is weird :S i have this error on version 11 I am going to take a look at it

**Review by nowakweronika — APPROVED** [review]

File: LICENSE (hunk 3 above)
Comment (trOnk12) [reply LICENSE]: mistake ;s thanks
